firefly_client/firefly_client.py

```diff
@@ -595,46 +597,77 @@ def show_table(self, file_on_server, tbl_id=None, title=None, page_size=100, is_
 
         return self.dispatch_remote_action(self.channel, FireflyClient.ACTION_DICT['ShowTable'], payload)
 
-    def show_xyplot(self, file_on_server, **additional_params):
+    def fetch_table(self, file_on_server, tbl_id=None, page_size=1):
         """
-        Show a XY plot
+        Fetch table data without showing them.
 
         Parameters
         ----------
         file_on_server : str, optional
-            The is the name of the file on the server.  If you use upload_file()
+            The name of the file on the server.  If you use upload_file()
             then it is the return value of the method. Otherwise it is a file that
             Firefly has direct access to.
-        **additional_params
-            Additional parameters for XY Plot, please see the details in 'Other Parameters'.
+        tbl_id : str, optional
+            A table ID. It will be created automatically if not specified.
+        page_size : int, optional
+            The number of rows to fetch.
 
-        Other Parameters
+        Returns
+        -------
+        out : dict
+            Status of the request, like {'success': true}. 
+        """
+
+        if not tbl_id:
+            tbl_id = FireflyClient._gen_item_id('Table')
+        tbl_req = {'startIdx': 0, 'pageSize': page_size, 'source': file_on_server,
+                   'id': 'IpacTableFromSource', 'tbl_id': tbl_id}
+        meta_info = {'title': tbl_id, 'tbl_id': tbl_id}
+        tbl_req.update({'META_INFO': meta_info})
+        payload = {'request': tbl_req, 'hlRowIdx': 0}
+        return self.dispatch_remote_action(self.channel, FireflyClient.ACTION_DICT['FetchTblData'], payload)
+
+    def show_xyplot(self, tbl_id, chart_params, standalone=False):
```

Comment:
To make more Pythonic, we would like to pass in optional keyword arguments. This can be done with the function signature as

    show_xyplot(self, tbl_id, standalone=False, **chart_params)

Then inside the function, `chart_params` will be a dictionary that will work the way you have coded.